### PR TITLE
`Quiz exercises`: Improve performance when saving submissions in synchronized quizzes

### PR DESCRIPTION
**Pull request by krusche**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I implemented the changes with a good performance and prevented too many database calls.

### Motivation and Context
There were unnecessary many DB operations when saving quiz submissions in the synchronized mode.
The PR also addresses an exceptional case, where Artemis might try to store participations + quiz submissions multiple times into the database, but cannot recover from it and would do this on a continuous basis

### Steps for Testing
Prerequisites:
- 1 Instructor
- Multiple Students
- 1 Quiz Exercise (synchronized)

1. Create the quiz
2. Participate in it with multiple students
3. Check that everything works correctly

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

**Review by Santia-go — APPROVED** [review]
Tested on TS1
Everything works as expected. No issues related to the PR were found.

**Review by SimonPaul02 — APPROVED** [review]
Tested on TS1, works as described

**Review by yuling-sun — DISMISSED** [review]
Tested on TS1. Worked as described.

File: src/main/java/de/tum/in/www1/artemis/service/scheduled/quiz/QuizScheduleService.java [hunk 1]
```diff
@@ -559,7 +557,7 @@ private void removeUnnecessaryObjectsBeforeSendingToClient(StudentParticipation
      * @param userSubmissionMap a Map with all submissions for the given quizExercise mapped by the username
      * @param userBatchMap      a Map of the username to quiz batch id for the given quizExercise
      * @param batchCache        a Map of all the batches for the given quizExercise
-     * @return the number of processed submissions (submit or timeout)
+     * @return                  the number of processed submissions (submit or timeout)
```
Comment (SimonPaul02) [inline src/main/java/de/tum/in/www1/artemis/service/scheduled/quiz/QuizScheduleService.java:560]: I don't understand what you have done here. It seems that you have added unnecessary whitespace

Comment (Strohgelaender) [reply src/main/java/de/tum/in/www1/artemis/service/scheduled/quiz/QuizScheduleService.java:560]: Every description (of the parameters and the return value) is now on the same vertical line.

**Review by TheTrueVester — APPROVED** [review]
Tested on ts1. Everything worked as expected.
